### ISSUE TITLE
Google Translate TTS doesn't support accents/dialects

### ISSUE THREAD

**Issue by tetsuo13** [tetsuo13/home-assistant-config#13]
At some point specifying the language value of "en-au" using Google Translate for TTS:

```
tts:
  - platform: google_translate
    language: 'en-au'
```

It's possible that https://github.com/home-assistant/core/issues/46107 details around the time when this feature stopped working. It notes that https://github.com/home-assistant/core/pull/46110 fixed the issue but that wasn't found to be the case. https://github.com/pndurette/gTTS/blob/v2.2.2/CHANGELOG.rst#deprecations-and-removals notes it was removed in the 2.2.2 version of the gTTS library.

As a workaround the Google Translate was downloaded and set up as a custom component and modified to support accents by supplying a hard-coded "tld" parameter in the `tts.py` file:

```python
def get_tts_audio(self, message, language, options=None):
    """Load TTS from google."""
    tts = gTTS(text=message, lang=language, tld='com.au')
    mp3_data = BytesIO()
```

This custom component has never been committed to the repo.

https://github.com/home-assistant/core/pull/81768 has potential to fix this issue once and for all.

**Comment by tetsuo13** [comment]
https://github.com/home-assistant/core/pull/81768 has been included in [Home Assistant Core 2023.1](https://www.home-assistant.io/changelogs/core-2023.1). The documentation for [Google Translate Text-to-Speech](https://www.home-assistant.io/integrations/google_translate/) now includes a note on the `tld` configuration variable.

**Comment by tetsuo13** [comment]
There wasn't anything to change since the 2023.1 release, dialects just started working again.

You can either use this method (which is what was already in place):

```yaml
tts:
  - platform: google_translate
    language: "en-au"
```

or

```yaml
tts:
  - platform: google_translate
    language: "en"
    options:
      tld: com.au
```
